server/etcdserver/raft.go

```diff
@@ -620,15 +620,19 @@ func restartAsStandaloneNode(cfg config.ServerConfig, snapshot *raftpb.Snapshot,
 		}
 	}
 
+	member := membership.Member{
+		ID:             id,
+		RaftAttributes: membership.RaftAttributes{PeerURLs: []string{"http://localhost:2380"}},
+	}
+	cl := membership.NewClusterSingleNodeCluster(cfg.Logger, &member, membership.WithMaxLearners(cfg.ExperimentalMaxLearners))
```

Comment:
I think this is a good approach, but we need to re-write the wal with the new metadata set. Otherwise it might happen that the member restarts and reads the old cluster id again

Reply:
or at least remove the WAL, so the snapshot becomes the authority again

Reply:
... or we can also try to append a new metadata record on the WAL same as createConfigChangeEnts:
> w.encoder.encode(&walpb.Record{Type: metadataType, Data: metadata})

and then the metadata should be pulled off from there. We just need to make sure the cid is computed uniquely enough

Reply:
because it's Friday, here's an entirely untested code path on this :rofl: 
https://github.com/openshift/etcd/compare/openshift-4.18...tjungblu:change_meta?expand=1

Reply:
Let's see how it works!